### PR TITLE
Update low level HTML pages to null safety

### LINKED ISSUE #2438: 'Remove DOM elements' page issue
Page URL: https://dart.dev/tutorials/web/low-level-html/remove-elements
Page source: https://github.com/dart-lang/site-www/tree/master/src/_tutorials/web/low-level-html/remove-elements.md

Description of issue:
CSS `font-size `property to make text larger is missing in dartpad code.

**The code in dartpad**
`#to-do-list li:hover {
  color: red;
  cursor: pointer;
}`
**The code in explanation**
`#to-do-list li:hover {
  color: red;
  font-size: 18px;
  cursor:pointer;
}`

### LINKED ISSUE #3415: Update low level HTML pages
- [ ] Update to null safety
- [ ] Utilize Effective Dart and recommended lint standards in code
- [ ] Update diagrams (consider SVG)
- [ ] Auto include code where possible
- [ ] Update text for changes in DartPad's UI
- [ ] Consider using embedded DartPads

### LINKED ISSUE #3697: [PAGE ISSUE] 'Add Elements to the DOM' Page Issue
### Page URL

https://dart.dev/tutorials/web/low-level-html/add-elements#:~:text=%20Add%20elements%20to%20the%20DOM%20%201,app.%20The%20main%20%28%29%20function%20uses...%20More%20

### Page source

_No response_

### Describe the problem

On the example with the todo list, it is currently not possible to run the app because there are many errors within the code. This could be because the code there is actually an older form of Dart. If you try to run the app, it will tell you that there was an error compiling to JavaScript because there are many errors in the code.

**### Errors in the code provided:**

- line 9 • The non-nullable variable 'toDoInput' must be initialized. (view docs)
  Try adding an initializer expression.

- line 10 • The non-nullable variable 'toDoList' must be initialized. (view docs)
  Try adding an initializer expression.
 
- line 11 • The non-nullable variable 'deleteAll' must be initialized. (view docs)
  Try adding an initializer expression.
 
- line 14 • A value of type 'Element?' can't be assigned to a variable of type 'InputElement'. (view docs)
  Try changing the type of the variable, or casting the right-hand type to 'InputElement'.
 
- line 15 • A value of type 'Element?' can't be assigned to a variable of type 'UListElement'. (view docs)
  Try changing the type of the variable, or casting the right-hand type to 'UListElement'.
 
- line 20 • Unnecessary new keyword. (view docs)

### Expected fix

Please could you fix/rewrite the code so that the app runs as expected?

### Additional context

There are no problems anywhere else on the page.

### PR DESCRIPTION
**Pull request by parlough**
**Connect Dart and HTML** | [Staged](https://dart-dev--pr3596-feature-update-low-l-yr9ue9le.web.app/tutorials/web/low-level-html/connect-dart-html)
- [x] Update code snippets to support null safety
- [x] Add any necessary mentions to null safety
- [x] Update the diagrams for most up to date text

**Add elements to the DOM** | [Staged](https://dart-dev--pr3596-feature-update-low-l-yr9ue9le.web.app/tutorials/web/low-level-html/add-elements)
- [x] Update code snippets
- [x] Add any necessary mentions to null safety
- [x] Update the diagrams for most up to date text

**Remove DOM elements** | [Staged](https://dart-dev--pr3596-feature-update-low-l-yr9ue9le.web.app/tutorials/web/low-level-html/remove-elements)
- [x] Update code snippets
- [x] Add any necessary mentions to null safety
- [x] Update the diagrams for most up to date text

Fixes #2438
Fixes #3697
Closes #3415

### CODE REVIEW

**Comment by parlough** [comment]
I've been receiving multiple DartPad errors related to embeds and HTML embeds that I will need to investigate and create appropriate issues or fix them myself.

**Comment by parlough** [comment]
Hey @domesticmouse, before I create updated diagrams for these pages, it would be great if you could look over the code changes in the embedded DartPads. I've kept them relatively minimal for these purposes but want to make sure I have them in a good direction before I dive fully into the updated diagrams. You can view the previews linked above to see and interact with them.

File: src/_tutorials/web/low-level-html/connect-dart-html.md [hunk 1]
```diff
@@ -239,60 +236,89 @@ making all of the classes and functions
...
-This app's main() function contains a single
+This app's `main()` function contains a single
 line of code that is a little like a run-on sentence
 with multiple things happening one after another.
 Let's deconstruct it.
 
-querySelector() is a top-level function provided by the Dart HTML library
-that gets an Element object from the DOM.
+`querySelector()` is a top-level function provided by the Dart HTML library
+that gets an `Element?` object from the DOM matching the passed in specifier.
```
Comment (kwalrath) [inline src/_tutorials/web/low-level-html/connect-dart-html.md:270]: I don't think adding more words to this sentence helps... it's already pretty long, and those details are covered below.

```suggestion
that gets an `Element?` object from the DOM.
```

File: src/_tutorials/web/low-level-html/connect-dart-html.md [hunk 2]
```diff
@@ -239,60 +236,89 @@ making all of the classes and functions
...
-<a href="#" class="dart-popover" data-toggle="popover" data-html="true" data-trigger="hover focus" data-content="Returns the matching Element">querySelector('#RipVanWinkle')</a>.text = 'Wake up, sleepy head!';
-</pre>
+{% prettify dart tag=pre+code %}
+[!querySelector!]('#RipVanWinkle')!.text = 'Wake up, sleepy head!';
+{% endprettify %}
 
-The argument to querySelector(), a string,
-is a CSS selector that identifies the object.
+In this example, the argument passed to the function, `'#RipVanWinkle'`,
+is a string literal containing the CSS selector that identifies the object.
```
Comment (kwalrath) [inline src/_tutorials/web/low-level-html/connect-dart-html.md:277]: I like something closer to the original wording, which is talking in general about the querySelector arg, not about the particular arg in this example.

```suggestion
The argument to `querySelector()` is a string
containing a CSS selector that identifies the object.
```

File: src/_tutorials/web/low-level-html/connect-dart-html.md [hunk 3]
```diff
@@ -370,7 +398,7 @@ on how to convert a DartPad app into one you can run in the browser.
 Here's the full HTML code for this app,
 assuming that the Dart code is in a file named `main.dart`:
 
-```dart
+{% prettify html tag=pre+code %}
```
Comment (kwalrath) [inline src/_tutorials/web/low-level-html/connect-dart-html.md:401]: Good catch!

Does  <code>```html</code> not work?

File: src/_tutorials/web/low-level-html/connect-dart-html.md [hunk 4]
```diff
@@ -394,26 +422,26 @@ Most HTML uses cascading style sheets (CSS) to define _styles_
...
 Let's customize the CSS for the mini app.
 
-1. Click **CSS**.
+1. Click **CSS**, at the upper left of DartPad.
    The view switches from Dart code to the (non-existent) CSS code.
 
 2. Add the following CSS code:
 
-   {% prettify none tag=pre+code %}
+   {% prettify css tag=pre+code %}
```
Comment (kwalrath) [inline src/_tutorials/web/low-level-html/connect-dart-html.md:430]: Looks much better with the formatting!

But the only time I can remember choosing to use prettify is when I use highlighting. (<code>```css</code> doesn't work?)

File: src/_tutorials/web/low-level-html/connect-dart-html.md [hunk 5]
```diff
@@ -431,21 +459,21 @@ a class, or an attribute.
...
-| ID selector | #RipVanWinkle | Matches a single, unique element |
-| HTML element | p | Matches all paragraphs |
-| HTML element | h1 | Matches all level-one headers |
-| Class | _.classname_ | Matches all items with the class _classname_  |
-| Asterisk | * | Matches all elements |
-| Attribute | input[type="button"] | Matches all button input elements |
+| ID selector | `#RipVanWinkle` | Matches a single, unique element |
+| HTML element | `p` | Matches all paragraphs |
+| HTML element | `h1` | Matches all level-one headers |
+| Class | `_.classname_` | Matches all items with the class _classname_  |
```
Comment (kwalrath) [inline src/_tutorials/web/low-level-html/connect-dart-html.md:474]: _.classname_ isn't rendering correctly in the second column. Maybe use a non-code font for everything after the period? The following may or may not work ;)
```suggestion
| Class | <code>.<em>classname</em></code> | Matches all items with the class _classname_  |
```

File: src/_tutorials/web/low-level-html/connect-dart-html.md [hunk 6]
```diff
@@ -467,19 +495,19 @@ A CSS rule has two main parts: a selector and a set of declarations.
 <img class="scale-img-max" src="/tutorials/web/images/css-rule-explained.png"
      alt="The parts of a CSS rule">
 
-In the mini app, the selector #RipVanWinkle is an ID selector,
-as signaled by the hash tag (#);
+In the mini app, the selector `#RipVanWinkle` is an ID selector,
+as signaled by the hashtag (`#`);
```
Comment (kwalrath) [inline src/_tutorials/web/low-level-html/connect-dart-html.md:499]: Good catch on _hashtag_!

**Review by kwalrath — COMMENTED** [review]
Finally getting to this review! Here's my feedback on connect-dart-html page. Now to look at the other pages..

File: diagrams/tutorials/web/css-rule-explained.drawio [hunk 7]
```diff
@@ -0,0 +1 @@
...0fyQ0TqqZbcu0pLjqbAx5giOxb5jQZoJ8KYQqS+nBJ0zHropLOe5qpLVemCRc/c4AJG8xDT7zWHEWfrv+8W3u3kznpZcHzHZmw2ax6rGKgBQ7HhHtxJqg9T6higQZDnXrHiAHW6JSBjUbilvB1RVOKdNwX4s7oYSxBmIni0GJUoCd46EVfGC1+qM75BexEDEjOKP5RSjSoiHMi65X29InFDteS8LYLtRzJcV9jZKje1DGfMGELHaCNvZ6s5mDHTMac7CFEDoi66FVVy44OF2b0BCpyGE05naNJJwAIlKiJCzTMgNctCiHGPYj19T3Ry4tTciTBo0qzmDD3rj2fAQYCgbjP8Dbfdd4V7gyslV9+H3PX22Wb4l2pQoV2jOvh7ZT9WnCvfBOh3v16dPuZhMEKPHJnn/8Z+P9/XXqOQN4z5gOTkQfoBirYuvommbfMb+h/B5oAOybr6t+MG+r68DoGU41LfhdnhV1SyM1zQuoAHQLYnhojH2xR0OKwueHkhYfJo6v6Yt5Ps2JpNsT5wHw1dTQSE9TK8QpTlMsY8qnSmSvFo87HN7HxUmdhiWPteOAYUXWBdQneW+4vAWW6VvrD/zNL8f6jiqN/TKlcawvgo8Ijl49iM2I1BRCU8vM0eG6ctiQDiMQHXHJYaWUx2BYQC0D7sGRJTKozdbTStOTlQHxeeJe6SiNa/WVZjGgNDZ6q5vFHlCaDt6QAWW6GO4ke1xLoDBRzwPfZklf9F9BuNHSa4cT9YUbDUXTfatojuu25nMrrLN/d6JqaIiu7YLI1I3VGfSDANqud1rl6zNauhw5pFoR24AMkvqJA1NrqMaIQmq9Mg0pjSI9zSAD2mnIOfOM1yCU3SHUsn8+h47nm51Or8enSxIyLLGigufvKQccS/rOlvPPrY42W33sl+fEftbDvlSBgDD4Y4bdv1/0z//H53VuZqBDD/36jeAs8C+fv5gJj1b6rUQHiOE8p+HYFTymxU9lYicDD63rpvsGA8iBqh96XReeqd2aNevy5cEsuag8tpKyfgpBot5T0LOoN1Ad+rGrbJJAZk4f2u6HkDYzfBW0uOUrSXHdTn7SIUteRNyMar4BdRzN3GXb0azz56ngV4WonqOCePW2T0gSfyNL/J+M/20yeotObmO9kIzuonNRetYrkRGqx3fVsvvxcRptfgE=</diagram></mxfile>
```
Comment (kwalrath) [inline diagrams/tutorials/web/css-rule-explained.drawio:1]: Let's use sentence case for labels:

* CSS Rule -> CSS rule
* CSS Selector -> CSS selector
* (Declarations is good as-is)

**Review by kwalrath — COMMENTED** [review]
Some feedback on figures (which look great, in general — much more legible!)

File: src/_tutorials/web/low-level-html/connect-dart-html.md (hunk 3 above)
Comment (parlough) [reply src/_tutorials/web/low-level-html/connect-dart-html.md:401]: It does, but the resulting HTML highlighting does not seem to handle it well at all:

image

For now, I'll stick to prettify for these and investigate this separately (https://github.com/dart-lang/site-www/issues/3685)

Comment (kwalrath) [reply src/_tutorials/web/low-level-html/connect-dart-html.md:401]: Ah right. I was encountering this myself recently.

**Comment by parlough** [comment]
I will opening a new PR targeting the `main` branch with these changes.

**Pull request closed without merging**